### PR TITLE
T5368: service ids ddos-protection add support sflow mode

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

sFlow mode requires fewer resources then mode `mirror`
Integrate it into configuration mode
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5368

## How to test
Get flows from eth0 interface and export them to FastNetmon
```
set system sflow interface 'eth0'
set system sflow interface 'eth1'
set system sflow server 127.0.0.1

set service ids ddos-protection direction 'in'
set service ids ddos-protection mode 'sflow'
set service ids ddos-protection network '192.0.2.0/24'
set service ids ddos-protection sflow listen-address '127.0.0.1'
set service ids ddos-protection sflow port '6343'
set service ids ddos-protection threshold general pps '10000'
```
Send traffic to 192.0.2.1  more than 10Kpps
fnm

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/service-ids-ddos-protection.xml.in [hunk 1]
```diff
@@ -70,17 +70,34 @@
...
-              <node name="mode">
+              <leafNode name="mode">
                 <properties>
                   <help>Traffic capture modes</help>
+                  <completionHelp>
+                    <list>mirror sflow</list>
+                  </completionHelp>
+                  <valueHelp>
+                    <format>mirror</format>
+                    <description>>Listen mirrored traffic mode</description>
```
Comment (dmbaturin) [inline interface-definitions/service-ids-ddos-protection.xml.in]: There's a stray closing `>` there. There's also missing preposition. I think "Listen to mirrored traffic" is a better wording.

File: interface-definitions/service-ids-ddos-protection.xml.in [hunk 2]
```diff
@@ -70,17 +70,34 @@
                   <multi/>
                 </properties>
               </leafNode>
-              <node name="mode">
+              <leafNode name="mode">
                 <properties>
                   <help>Traffic capture modes</help>
```
Comment (dmbaturin) [inline interface-definitions/service-ids-ddos-protection.xml.in]: We usually use singulars in command descriptions< "Traffic capture _mode_".

File: src/conf_mode/service_ids_fastnetmon.py [hunk 3]
```diff
@@ -55,8 +55,11 @@ def verify(fastnetmon):
     if 'mode' not in fastnetmon:
         raise ConfigError('Specify operating mode!')
 
-    if 'listen_interface' not in fastnetmon:
-        raise ConfigError('Specify interface(s) for traffic capture')
+    if fastnetmon.get('mode') == 'mirror' and 'listen_interface' not in fastnetmon:
+        raise ConfigError('Specify interface(s) for traffic capture mode mirror')
```
Comment (dmbaturin) [inline src/conf_mode/service_ids_fastnetmon.py]: This is a pretty confusing error message. What's a "traffic capture mode mirror"? A magical artifact ­— a mirror where you can see the traffic capture mode setting from your config at midnight under full moon? :)

I think it should be "Incorrect settings for 'mode mirror": must specify interface(s) for traffic mirroring".

File: src/conf_mode/service_ids_fastnetmon.py [hunk 4]
```diff
@@ -55,8 +55,11 @@ def verify(fastnetmon):
...
+
+    if fastnetmon.get('mode') == 'sflow' and 'listen_address' not in fastnetmon.get('sflow', {}):
+        raise ConfigError('Specify sflow "listen-address" for mode sflow')
```
Comment (dmbaturin) [inline src/conf_mode/service_ids_fastnetmon.py]: I'd say "Incorrect settings for 'mode sflow': must specify sFlow listen address".

**Review by dmbaturin — CHANGES_REQUESTED** [review]
I agree with the idea, but there seems to be a stray angle bracket and some messages can be improved.

File: interface-definitions/service-ids-ddos-protection.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/service-ids-ddos-protection.xml.in]: Fixed! thanks

File: interface-definitions/service-ids-ddos-protection.xml.in (hunk 2 above)
Comment (sever-sever) [reply interface-definitions/service-ids-ddos-protection.xml.in]: Fixed

File: src/conf_mode/service_ids_fastnetmon.py (hunk 3 above)
Comment (sever-sever) [reply src/conf_mode/service_ids_fastnetmon.py]: Done!

File: src/conf_mode/service_ids_fastnetmon.py (hunk 4 above)
Comment (sever-sever) [reply src/conf_mode/service_ids_fastnetmon.py]: Done!

**Review by dmbaturin — APPROVED** [review]

**Pull request merged**